### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by HassanNiazi**
Authentication Module - Missing services

### CODE REVIEW

File: src/models/login.model.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export class Login {
     constructor(
-        email: string = null,
-        password: string = null
+       public email: string = null,
+       public password: string = null
     ) { }
```
Comment (Abdullah2993) [inline src/models/login.model.ts]: Add interface and preferably use params method for constructor
```
export interface ILogin {
    email: string
    password: string
}

export class Login implements ILogin {
    email: string;
    password: string;

    constructor(params?: ILogin) {
        if(params){
            this.email = params.email;
            this.password = params.password;
        }
     }
}
```

File: src/models/register.model.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 export class Register {
     constructor(
```
Comment (Abdullah2993) [inline src/models/register.model.ts]: same as above

File: src/models/request-reset.model.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export class RequestResetPassword {
+    constructor(public email: string = null) { }
```
Comment (Abdullah2993) [inline src/models/request-reset.model.ts]: same as above and no need use null use `?` like above

File: src/models/reset-password.model.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 export class ResetPassword {
     constructor(
```
Comment (Abdullah2993) [inline src/models/reset-password.model.ts]: same

File: src/app/auth/request-reset-password/request-reset-password.component.ts [hunk 5]
```diff
@@ -7,9 +9,16 @@ import { Component, OnInit } from '@angular/core';
 })
 export class RequestResetPasswordComponent implements OnInit {
 
+  model: RequestResetPassword = new RequestResetPassword();
   constructor() { }
 
+  emailPattern = EMAIL_REGEX;
```
Comment (Abdullah2993) [inline src/app/auth/request-reset-password/request-reset-password.component.ts]: constant?

File: src/app/auth/login-page/login-page.component.ts [hunk 6]
```diff
@@ -8,13 +18,21 @@ import { Login } from '../../../models/login.model';
...
+  emailPattern = EMAIL_REGEX;
+  passwordPattern = PASSWORD_REGEX;
   model: Login;
 
   constructor() { }
 
   ngOnInit() {
     if (!this.model) {
-      this.model = new Login(null, null);
+      this.model = new Login();
```
Comment (Abdullah2993) [inline src/app/auth/login-page/login-page.component.ts]: why?

File: src/app/auth/login-page/login-page.component.ts [hunk 7]
```diff
@@ -8,13 +18,21 @@ import { Login } from '../../../models/login.model';
 })
 export class LoginPageComponent implements OnInit {
 
+  emailError = EMAIL_ERROR;
```
Comment (Abdullah2993) [inline src/app/auth/login-page/login-page.component.ts]: constants?

File: src/models/forgot-password.model.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export class ForgotPassword {
```
Comment (Abdullah2993) [inline src/models/forgot-password.model.ts]: Use interface and params

File: src/app/auth/request-reset-password/request-reset-password.component.ts (hunk 5 above)
Comment (Abdullah2993) [reply src/app/auth/request-reset-password/request-reset-password.component.ts]: Btw is there any better way to use constants in templates?

**Comment by HassanNiazi** [comment]
Updated the models approach and DI for the constants.

**Pull request merged**